container/app/entrypoint.py

```diff
@@ -0,0 +1,70 @@
+#!/venv/bin/python3
+#
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import argparse
+import json
+import os
+import subprocess
+
+from ovirt_imageio.client import _utils
+from ovirt_imageio._internal import qemu_img
+from ovirt_imageio._internal import qemu_nbd
+from ovirt_imageio._internal import nbd
+
+TICKET_PATH = "/app/ticket.json"
+OVIRT_IMAGEIO = "/venv/bin/ovirt-imageio"
+
+
+def _generate_ticket(uuid, socket, size):
+    ticket_info = {}
+    ticket_info["uuid"] = uuid
+    ticket_info["url"] = f"nbd:unix:{socket}"
+    ticket_info["size"] = size
+    ticket_info["timeout"] = 3000
+    ticket_info["inactivity_timeout"] = 300
+    ticket_info["ops"] = ["read"]
+    return ticket_info
+
+
+def create_ticket(ticket_id, image, socket):
+    image_size = qemu_img.info(image)["virtual-size"]
+    ticket_info = _generate_ticket(ticket_id, socket, image_size)
+    with open(TICKET_PATH, "w+") as f:
+        f.write(json.dumps(ticket_info))
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="ovirt-imageio")
+    parser.add_argument("image", help="Path to image.")
+    parser.add_argument(
+        "--ticket-id",
+        help="Optional. Set the ID of the ticket. It is recommended "
+        "to use with random string for better security. "
+        "Defaults to the image filename.")
+
+    return parser.parse_args()
+
```

Comment:
We can implement the default value of args.ticket_id here. This is little cleaner
since we document the default here:

    args = parser.parse_args()
    args.ticket_id = args.ticket_id or os.path.basename(args.image)
    return args

Now we can pass args to the rest of the function since args.image and args.ticket_id are always complete. I think we use this same approach in ovirt-img
when we pass args initialized from both command line and config file. The rest of the application should not care how args was initialized.